Makefile

```diff
@@ -345,6 +346,10 @@ API.md: cmd/podman/varlink/io.podman.varlink
 
 validate: gofmt .gitvalidation
 
+build-all-new-commits:
+	# Validate that all the commits build on top of $(GIT_BASE_BRANCH)
+	git rebase $(GIT_BASE_BRANCH) -x make
+
```

Comment:
I realize you're still working on this, but just quickly:  I think it's probably best to leave the repository management up to outside forces.  Otherwise the Makefile is assuming too much responsibility over the state/condition of the source.  

Having the source-management outside also makes adding important logic a lot easier.  For example, perhaps the test should immediately ``exit(0)`` with a warning if it finds a commits containing `fixup!`, `squash!`, or `DO NOT MERGE`. i.e. before it even starts building.  On the other hand, if ``$CIRRUS_BRANCH == 'master'`` (post-merge testing) then the test should throw a very large, red error message, and `exit(1)`.

Make sense?